### ISSUE TITLE
websocket terminated due to fail in matching function

### ISSUE THREAD

**Issue by wasdee** [emqx/emqx#13744]
### What happened?

my mqtt client with transport protocal websocket keep disconnecting every sec.

## looking at log
`websocket_terminated, reason: {crash,error,{badmatch,25391}}, username: user_a`
`2024-09-02T17:01:44.287325+00:00 [error] crasher: initial call: cowboy_clear:connection_process/4, pid: <0.6171.0>, registered_name: [], error: {{badmatch,25391},[{emqx_ws_connection,handle_outgoing,2,[{file,"emqx_ws_connection.erl"},{line,814}]},{emqx_ws_connection,websocket_info,2,[{file,"emqx_ws_connection.erl"},{line,998}]},{cowboy_websocket,handler_call,6,[{file,"cowboy_websocket.erl"},{line,487}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,251}]}]}, ancestors: [<0.2967.0>,<0.2966.0>,ranch_sup,<0.2519.0>], message_queue_len: 0, messages: [], links: [#Port<0.508>,<0.2967.0>], dictionary: [{{#{retain => true,qos => 1,action_type => publish},<<"services/control">>},{allow,1725296493956}},{send_pkt,242},{{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/profile">>},{allow,1725296493962}},{authz_keys_q,{[{#{retain => false,qos => 0,action_type => publish},<<"brain">>},{#{retain => true,qos => 1,action_type => publish},<<"services/control/actual_state">>},{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/profile">>},{#{qos => 0,action_type => subscribe},<<"services/brain/cam">>},{#{qos => 0,action_type => subscribe},<<"services/brain/config/working_envelop/y_floor">>},{#{qos => 0,action_type => subscribe},<<"services/brain/config/working_envelop/x_far">>},{#{qos => 0,action_type => subscribe},<<"services/brain/config/point_cloud_transformation">>},{#{qos => 0,action_type => subscribe},<<"services/emer/emergency">>},{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/rotate">>},{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/sling">>},{#{qos => 0,action_type => subscribe},<<"services/control_center/desired_state">>},{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/ready">>},{#{qos => 0,action_type => subscribe},<<"MCU_encoder/acknowledge">>},{#{qos => 0,action_type => subscribe},<<"MCU_encoder/state">>},{#{retain => true,qos => 1,action_type => publish},<<"services/control">>}],[{#{retain => true,qos => 1,action_type => publish},<<"services/control/log">>}]}},{{#{qos => 0,action_type => subscribe},<<"services/brain/cam">>},{allow,1725296493962}},{{#{retain => false,qos => 0,action_type => publish},<<"brain">>},{allow,1725296494041}},{{#{qos => 0,action_type => subscribe},<<"services/emer/emergency">>},{allow,1725296493961}},{{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/ready">>},{allow,1725296493957}},{{#{qos => 0,action_type => subscribe},<<"services/brain/config/working_envelop/y_floor">>},{allow,1725296493962}},{recv_msg,230},{{#{qos => 0,action_type => subscribe},<<"MCU_encoder/state">>},{allow,1725296493956}},{send_oct,25391},{recv_pkt,243},{send_cnt,242},{'recv_msg.qos0',102},{{#{qos => 0,action_type => subscribe},<<"services/brain/config/point_cloud_transformation">>},{allow,1725296493961}},{{#{retain => true,qos => 1,action_type => publish},<<"services/control/actual_state">>},{allow,1725296493963}},{recv_oct,56072},{'recv_msg.qos1',128},{incoming_bytes,7098},{authz_cache_size,16},{send_msg,101},{recv_cnt,243},{'$logger_metadata$',#{peername => "172.18.0.1:48458",username => <<"user_a">>,clientid => <<"control">>}},{'send_msg.qos0',101},{{#{qos => 0,action_type => subscribe},<<"services/brain/config/working_envelop/x_far">>},{allow,1725296493961}},{{#{qos => 0,action_type => subscribe},<<"MCU_encoder/acknowledge">>},{allow,1725296493957}},{outgoing_pubs,0},{guid,{1725296504192270,207584359356443,232}},{{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/sling">>},{allow,1725296493958}},{{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/rotate">>},{allow,1725296493958}},{{#{retain => true,qos => 1,action_type => publish},<<"services/control/log">>},{allow,1725296493955}},{outgoing_bytes,0},{incoming_pubs,28},{{#{qos => 0,action_type => subscribe},<<"services/control_center/desired_state">>},{allow,1725296493957}}], trap_exit: true, status: running, heap_size: 4185, stack_size: 28, reductions: 781474; neighbours:`

ai answer that
<details>
The error message you provided indicates a crash in the WebSocket connection process within the EMQX broker. The specific error is a `badmatch` error, which occurs when a pattern match fails in Erlang. Here’s a breakdown of the relevant parts of the error message:

1. **Error Details**:
   - **Error Type**: `badmatch`
   - **Value**: `25391`
   - **Module**: `emqx_ws_connection`
   - **Function**: `handle_outgoing/2`
   - **File**: `emqx_ws_connection.erl`
   - **Line**: 814

2. **Call Stack**:
   - `emqx_ws_connection:handle_outgoing/2` at line 814
   - `emqx_ws_connection:websocket_info/2` at line 998
   - `cowboy_websocket:handler_call/6` at line 487
   - `proc_lib:wake_up/3` at line 251

3. **Process Information**:
   - **Initial Call**: `cowboy_clear:connection_process/4`
   - **PID**: `<0.6171.0>`
   - **Ancestors**: `[<0.2967.0>, <0.2966.0>, ranch_sup, <0.2519.0>]`
   - **Message Queue Length**: 0
   - **Links**: `[#Port<0.508>, <0.2967.0>]`
   - **Dictionary**: Contains various MQTT topics and permissions

### Possible Causes and Solutions

1. **Pattern Matching Issue**:
   - The `badmatch` error suggests that the value `25391` did not match the expected pattern in the `handle_outgoing/2` function. This could be due to an unexpected value or state in the WebSocket connection.

2. **Resource Limits**:
   - The value `25391` might be related to a system resource limit, such as the number of open file descriptors or network sockets. Ensure that your system has sufficient resources and that limits are appropriately configured.

3. **Code Review**:
   - Review the code in the `emqx_ws_connection.erl` file, particularly around line 814, to understand what pattern is expected and why the value `25391` is causing a mismatch.

4. **Configuration Issues**:
   - Check the EMQX configuration for any misconfigurations that might lead to unexpected values being processed.

5. **Logs and Traces**:
   - Enable detailed logging and tracing in EMQX to gather more information about the state of the system when the error occurs. This can help identify the root cause.

### Example Code Review

Here’s a hypothetical example of what the code around line 814 in `emqx_ws_connection.erl` might look like:

```erlang
handle_outgoing(State) ->
    case State#state.some_field of
        ExpectedValue -> 
            %% Handle the expected value
            ok;
        _ ->
            %% This is where the badmatch error might occur
            {error, badmatch}
    end.
```

In this example, if `State#state.some_field` does not match `ExpectedValue`, a `badmatch` error will occur. You might need to add additional pattern matching or error handling to address unexpected values.

### Next Steps

1. **Increase System Limits**:
   - If the issue is related to system limits, increase the relevant limits (e.g., file descriptors, network sockets).

2. **Review and Update Code**:
   - Review the code in `emqx_ws_connection.erl` and add appropriate error handling or pattern matching to handle unexpected values.

3. **Check Configuration**:
   - Ensure that the EMQX configuration is correct and does not contain any misconfigurations that could lead to this issue.

4. **Enable Detailed Logging**:
   - Enable detailed logging and tracing in EMQX to gather more information about the error.

By following these steps, you should be able to identify and resolve the cause of the `badmatch` error in your WebSocket connection process.
</details>

### What did you expect to happen?

no websocket terminated

### How can we reproduce it (as minimally and precisely as possible)?

idk

### Anything else we need to know?

on version 5.7.2 this is not happen.

### EMQX version

<details>

```console
$ ./bin/emqx_ctl broker
# paste output here
```

</details>
5.8.0

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here
```

</details>
emqx:emqx  docker

### Log files

this is a log from log trace.
<details>
2024-09-02T17:01:44.192287+00:00 [PUBLISH] control@172.18.0.1:48458 msg: publish_to, topic: brain, username: user_a, payload: {"digital_twin_info":{"arm_pitch":23.0,"camera_pitch":-17.0,"camera_position":{"x":-3.5929717888724224,"y":10.758070179362745,"z":-0.5944170938379638},"lower_pulley":{"x":-1.1030629305556563,"y":9.476674815079745},"upper_pulley":{"x":-14.924898610346144,"y":15.27844302848228}},"autowaypoint_info":{"working_area":{"x":14.0,"y":3.0},"translation":{"x":0.4,"y":0.2},"scale":{"x":1.0,"y":1.0}}}
2024-09-02T17:01:44.286964+00:00 [WS-MQTT] control@172.18.0.1:48458 msg: mqtt_packet_sent, packet: PUBLISH(Q0, R0, D0, Topic=MCU_encoder/acknowledge, PacketId=undefined, Payload={"sling_position_pullIn":0,"sling_length_pullIn":0,"sling_velocity_pullIn":0,"sling_position_pullUpOut":0,"sling_length_pullUpOut":0,"sling_velocity_pullUpOut":0,"rotate_position":0,"angular_velocity":0,"heading":0}), username: user_a
2024-09-02T17:01:44.287115+00:00 [SOCKET] control@172.18.0.1:48458 msg: websocket_terminated, reason: {crash,error,{badmatch,25391}}, username: user_a
2024-09-02T17:01:44.287193+00:00 [PUBLISH] control@172.18.0.1:48458 msg: publish_to, topic: services/control, username: user_a, payload: offline
2024-09-02T17:01:44.287325+00:00 [error] crasher: initial call: cowboy_clear:connection_process/4, pid: <0.6171.0>, registered_name: [], error: {{badmatch,25391},[{emqx_ws_connection,handle_outgoing,2,[{file,"emqx_ws_connection.erl"},{line,814}]},{emqx_ws_connection,websocket_info,2,[{file,"emqx_ws_connection.erl"},{line,998}]},{cowboy_websocket,handler_call,6,[{file,"cowboy_websocket.erl"},{line,487}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,251}]}]}, ancestors: [<0.2967.0>,<0.2966.0>,ranch_sup,<0.2519.0>], message_queue_len: 0, messages: [], links: [#Port<0.508>,<0.2967.0>], dictionary: [{{#{retain => true,qos => 1,action_type => publish},<<"services/control">>},{allow,1725296493956}},{send_pkt,242},{{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/profile">>},{allow,1725296493962}},{authz_keys_q,{[{#{retain => false,qos => 0,action_type => publish},<<"brain">>},{#{retain => true,qos => 1,action_type => publish},<<"services/control/actual_state">>},{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/profile">>},{#{qos => 0,action_type => subscribe},<<"services/brain/cam">>},{#{qos => 0,action_type => subscribe},<<"services/brain/config/working_envelop/y_floor">>},{#{qos => 0,action_type => subscribe},<<"services/brain/config/working_envelop/x_far">>},{#{qos => 0,action_type => subscribe},<<"services/brain/config/point_cloud_transformation">>},{#{qos => 0,action_type => subscribe},<<"services/emer/emergency">>},{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/rotate">>},{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/sling">>},{#{qos => 0,action_type => subscribe},<<"services/control_center/desired_state">>},{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/ready">>},{#{qos => 0,action_type => subscribe},<<"MCU_encoder/acknowledge">>},{#{qos => 0,action_type => subscribe},<<"MCU_encoder/state">>},{#{retain => true,qos => 1,action_type => publish},<<"services/control">>}],[{#{retain => true,qos => 1,action_type => publish},<<"services/control/log">>}]}},{{#{qos => 0,action_type => subscribe},<<"services/brain/cam">>},{allow,1725296493962}},{{#{retain => false,qos => 0,action_type => publish},<<"brain">>},{allow,1725296494041}},{{#{qos => 0,action_type => subscribe},<<"services/emer/emergency">>},{allow,1725296493961}},{{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/ready">>},{allow,1725296493957}},{{#{qos => 0,action_type => subscribe},<<"services/brain/config/working_envelop/y_floor">>},{allow,1725296493962}},{recv_msg,230},{{#{qos => 0,action_type => subscribe},<<"MCU_encoder/state">>},{allow,1725296493956}},{send_oct,25391},{recv_pkt,243},{send_cnt,242},{'recv_msg.qos0',102},{{#{qos => 0,action_type => subscribe},<<"services/brain/config/point_cloud_transformation">>},{allow,1725296493961}},{{#{retain => true,qos => 1,action_type => publish},<<"services/control/actual_state">>},{allow,1725296493963}},{recv_oct,56072},{'recv_msg.qos1',128},{incoming_bytes,7098},{authz_cache_size,16},{send_msg,101},{recv_cnt,243},{'$logger_metadata$',#{peername => "172.18.0.1:48458",username => <<"user_a">>,clientid => <<"control">>}},{'send_msg.qos0',101},{{#{qos => 0,action_type => subscribe},<<"services/brain/config/working_envelop/x_far">>},{allow,1725296493961}},{{#{qos => 0,action_type => subscribe},<<"MCU_encoder/acknowledge">>},{allow,1725296493957}},{outgoing_pubs,0},{guid,{1725296504192270,207584359356443,232}},{{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/sling">>},{allow,1725296493958}},{{#{qos => 0,action_type => subscribe},<<"services/control_center/calibration/rotate">>},{allow,1725296493958}},{{#{retain => true,qos => 1,action_type => publish},<<"services/control/log">>},{allow,1725296493955}},{outgoing_bytes,0},{incoming_pubs,28},{{#{qos => 0,action_type => subscribe},<<"services/control_center/desired_state">>},{allow,1725296493957}}], trap_exit: true, status: running, heap_size: 4185, stack_size: 28, reductions: 781474; neighbours:
2024-09-02T17:01:45.291839+00:00 [MQTT] control@172.18.0.1:54986 msg: mqtt_packet_received, packet: CONNECT(Q0, R0, D0, ClientId=control, ProtoName=MQTT, ProtoVsn=5, CleanStart=false, KeepAlive=60, Username=user_a, Password=******, Will(Q1, R1, Topic=services/control Payload=offline)), username: user_a
2024-09-02T17:01:45.292001+00:00 [AUTHN] control@172.18.0.1:54986 msg: authenticator_result, authenticator: password_based:built_in_database, result: {ok,#{is_superuser => true}}, username: user_a
</details>

**Comment by alexdelprete** [comment]
I have the same issue:

```
      in call from emqx_ws_connection:websocket_info/2 (emqx_ws_connection.erl, line 998)
      in call from cowboy_websocket:handler_call/6 (cowboy_websocket.erl, line 487)
2024-09-03T02:50:49.307145+02:00 [error] Ranch listener 'ws:default' had connection process started with cowboy_clear:start_link/4 at <0.3508.0> exit with reason: {{badmatch,41217},[{emqx_ws_connection,handle_outgoing,2,[{file,"emqx_ws_connection.erl"},{line,814}]},{emqx_ws_connection,websocket_info,2,[{file,"emqx_ws_connection.erl"},{line,998}]},{cowboy_websocket,handler_call,6,[{file,"cowboy_websocket.erl"},{line,487}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,251}]}]}
      in call from emqx_ws_connection:websocket_info/2 (emqx_ws_connection.erl, line 998)
      in call from cowboy_websocket:handler_call/6 (cowboy_websocket.erl, line 487)
2024-09-03T02:50:50.373586+02:00 [error] Ranch listener 'ws:default' had connection process started with cowboy_clear:start_link/4 at <0.3512.0> exit with reason: {{badmatch,31606},[{emqx_ws_connection,handle_outgoing,2,[{file,"emqx_ws_connection.erl"},{line,814}]},{emqx_ws_connection,websocket_info,2,[{file,"emqx_ws_connection.erl"},{line,998}]},{cowboy_websocket,handler_call,6,[{file,"cowboy_websocket.erl"},{line,487}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,251}]}]}
      in call from emqx_ws_connection:websocket_info/2 (emqx_ws_connection.erl, line 998)
      in call from cowboy_websocket:handler_call/6 (cowboy_websocket.erl, line 487)
2024-09-03T02:50:51.450621+02:00 [error] Ranch listener 'ws:default' had connection process started with cowboy_clear:start_link/4 at <0.3522.0> exit with reason: {{badmatch,103121},[{emqx_ws_connection,handle_outgoing,2,[{file,"emqx_ws_connection.erl"},{line,814}]},{emqx_ws_connection,websocket_info,2,[{file,"emqx_ws_connection.erl"},{line,998}]},{cowboy_websocket,handler_call,6,[{file,"cowboy_websocket.erl"},{line,487}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,251}]}]}
      in call from emqx_ws_connection:websocket_info/2 (emqx_ws_connection.erl, line 998)
      in call from cowboy_websocket:handler_call/6 (cowboy_websocket.erl, line 487)
2024-09-03T02:50:52.570147+02:00 [error] Ranch listener 'ws:default' had connection process started with cowboy_clear:start_link/4 at <0.3526.0> exit with reason: {{badmatch,103121},[{emqx_ws_connection,handle_outgoing,2,[{file,"emqx_ws_connection.erl"},{line,814}]},{emqx_ws_connection,websocket_info,2,[{file,"emqx_ws_connection.erl"},{line,998}]},{cowboy_websocket,handler_call,6,[{file,"cowboy_websocket.erl"},{line,487}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,251}]}]}
      in call from emqx_ws_connection:websocket_info/2 (emqx_ws_connection.erl, line 998)
      in call from cowboy_websocket:handler_call/6 (cowboy_websocket.erl, line 487)
2024-09-03T02:50:53.652863+02:00 [error] Ranch listener 'ws:default' had connection process started with cowboy_clear:start_link/4 at <0.3530.0> exit with reason: {{badmatch,103121},[{emqx_ws_connection,handle_outgoing,2,[{file,"emqx_ws_connection.erl"},{line,814}]},{emqx_ws_connection,websocket_info,2,[{file,"emqx_ws_connection.erl"},{line,998}]},{cowboy_websocket,handler_call,6,[{file,"cowboy_websocket.erl"},{line,487}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,251}]}]}
      in call from emqx_ws_connection:websocket_info/2 (emqx_ws_connection.erl, line 998)
      in call from cowboy_websocket:handler_call/6 (cowboy_websocket.erl, line 487)
2024-09-03T02:50:54.736087+02:00 [error] Ranch listener 'ws:default' had connection process started with cowboy_clear:start_link/4 at <0.3534.0> exit with reason: {{badmatch,60242},[{emqx_ws_connection,handle_outgoing,2,[{file,"emqx_ws_connection.erl"},{line,814}]},{emqx_ws_connection,websocket_info,2,[{file,"emqx_ws_connection.erl"},{line,998}]},{cowboy_websocket,handler_call,6,[{file,"cowboy_websocket.erl"},{line,487}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,251}]}]}
      in call from emqx_ws_connection:websocket_info/2 (emqx_ws_connection.erl, line 998)
      in call from cowboy_websocket:handler_call/6 (cowboy_websocket.erl, line 487)
2024-09-03T02:50:55.802630+02:00 [error] Ranch listener 'ws:default' had connection process started with cowboy_clear:start_link/4 at <0.3538.0> exit with reason: {{badmatch,40897},[{emqx_ws_connection,handle_outgoing,2,[{file,"emqx_ws_connection.erl"},{line,814}]},{emqx_ws_connection,websocket_info,2,[{file,"emqx_ws_connection.erl"},{line,998}]},{cowboy_websocket,handler_call,6,[{file,"cowboy_websocket.erl"},{line,487}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,251}]}]}
      in call from emqx_ws_connection:websocket_info/2 (emqx_ws_connection.erl, line 998)
      in call from cowboy_websocket:handler_call/6 (cowboy_websocket.erl, line 487)
2024-09-03T02:50:56.855843+02:00 [error] Ranch listener 'ws:default' had connection process started with cowboy_clear:start_link/4 at <0.3542.0> exit with reason: {{badmatch,47546},[{emqx_ws_connection,handle_outgoing,2,[{file,"emqx_ws_connection.erl"},{line,814}]},{emqx_ws_connection,websocket_info,2,[{file,"emqx_ws_connection.erl"},{line,998}]},{cowboy_websocket,handler_call,6,[{file,"cowboy_websocket.erl"},{line,487}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,251}]}]}

```

**Comment by fisheralpha** [comment]
Websocket disconnects after about 100 seconds, ：

```
2024-09-03T11:16:42.712509+08:00 [error] crasher: initial call: cowboy_clear:connection_process/4, pid: <0.21501.0>, registered_name: [], error: {{badmatch,6811},[{emqx_ws_connection,handle_outgoing,2,[{file,"emqx_ws_connection.erl"},{line,814}]},{emqx_ws_connection,websocket_info,2,[{file,"emqx_ws_connection.erl"},{line,998}]},{cowboy_websocket,handler_call,6,[{file,"cowboy_websocket.erl"},{line,487}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,251}]}]}, ancestors: [<0.2725.0>,<0.2724.0>,ranch_sup,<0.2493.0>], message_queue_len: 0, messages: [], links: [#Port<0.695>,<0.2725.0>], dictionary: [{send_cnt,104},{recv_oct,47},{recv_cnt,3},{recv_pkt,3},{guid,{1725333302328295,268577189942269,1}},{outgoing_pubs,0},{incoming_bytes,47},{send_pkt,104},{outgoing_bytes,0},{send_oct,6811},{'$logger_metadata$',#{peername => "20.150.217.212:52388",clientid => <<"emqx_MDQ0Nj">>}},{{#{action_type => subscribe,qos => 0},<<"/#">>},{allow,1725333302663}},{authz_keys_q,{[{#{action_type => subscribe,qos => 0},<<"/#">>}],[]}},{send_msg,101},{'send_msg.qos0',101},{authz_cache_size,1}], trap_exit: true, status: running, heap_size: 987, stack_size: 28, reductions: 61168; neighbours:
2024-09-03T11:16:42.713516+08:00 [error] Ranch listener 'ws:default' had connection process started with cowboy_clear:start_link/4 at <0.21501.0> exit with reason: {{badmatch,6811},[{emqx_ws_connection,handle_outgoing,2,[{file,"emqx_ws_connection.erl"},{line,814}]},{emqx_ws_connection,websocket_info,2,[{file,"emqx_ws_connection.erl"},{line,998}]},{cowboy_websocket,handler_call,6,[{file,"cowboy_websocket.erl"},{line,487}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,251}]}]}
```

**Comment by id** [comment]
Can't reproduce this with paho.mqtt python lib.

Which mqtt client and MQTT version are you using @wasdee @alexdelprete @fisheralpha ? Could you share a sample code?

**Comment by alexdelprete** [comment]
I use Home Assistant, configured with WS, and after upgrade it didn't work. I switched to TCP and it worked. It has always worked in WS until 5.8.0.

I also tried with MQTT Explorer: configured a TCP connection and works perfectly, when I switched it to WS, it works but it disconnects every 3s.

Try MQTT Explorer.

**Comment by id** [comment]
In 5.8.0 we disabled default ACL rule which allowed subscribing on `#`. When I use MQTT Explorer, I see this:

```
2024-09-03T11:27:55.911331+00:00 [warning] tag: AUTHZ, clientid: mqtt-explorer-4d218604, msg: authorization_permission_denied, peername: 192.168.65.1:29993, topic: #, source: file, action: SUBSCRIBE(Q0)
```

If you really need this functionality, you can enable it in acl.conf

```
{allow, all, all, ["$SYS/#", "#"]}.
```

**Comment by id** [comment]
But other than this, it works just fine for me.

**Comment by id** [comment]
Found the bug. Thanks for the report guys. Fix is on its way.

**Comment by id** [comment]
Fix will be released in 5.8.1
